pkg/slo-controller/noderesource/plugins/batchresource/plugin.go

```diff
@@ -114,6 +114,38 @@ func (p *Plugin) Prepare(_ *configuration.ColocationStrategy, node *corev1.Node,
 		prepareNodeForResource(node, nr, resourceName)
 	}
 
+	// set origin batch allocatable
+	batchMilliCPU := util.GetBatchMilliCPUFromResourceList(node.Status.Allocatable)
+	batchMemory := util.GetBatchMemoryFromResourceList(node.Status.Allocatable)
+	originBatchAllocatable := corev1.ResourceList{
+		extension.BatchCPU:    *resource.NewQuantity(util.MaxInt64(batchMilliCPU, 0), resource.DecimalSI),
+		extension.BatchMemory: *resource.NewQuantity(util.MaxInt64(batchMemory, 0), resource.BinarySI),
+	}
+	if err := slov1alpha1.SetOriginExtendedAllocatableRes(node.Annotations, originBatchAllocatable); err != nil {
+		return err
+	}
+
+	if batchMilliCPU < 0 || batchMemory < 0 {
+		// batch resources are reset, no need to recalculate node status
+		return nil
+	}
+
+	// subtract third party allocated
+	thirdPartyBatchAllocated, err := slov1alpha1.GetThirdPartyAllocatedResByPriority(node.Annotations, extension.PriorityBatch)
```

Comment:
do we need to watch the change in third-party allocation?

Reply:
maybe later, now we just use the default ticker for updating.